### PR TITLE
feat: lots of RunIt form cleanup

### PR DESCRIPTION
**Pull request by jkaster**
Made some changes to the **RunIt** form, both for layout and editing various values
 
## Bug fixes

- enum values are now a string input instead of `{}`. There is an outstanding task to support a dropdown picker for enums that will come with the next significant change for the form editor
- DelimArray values are now a string input instead of `{}`

## Design improvements

- all inputs are on a single line now, saving form space. This is particularly useful for endpoints with many input values

image

- date time picker is now a pop over with a button to choose the date (but still doesn't have time picking)

image

- the default action for the form is now the first button, making it more convenient for tabbing through

image

### CODE REVIEW

**Review by josephaxisa — CHANGES_REQUESTED** [review]
Looking great, liking these changes! While you're at it, could you please add unit tests to:
1) ensure that when a date is picked, the corresponding state param value is what you'd expect it to be 
2) enum values are string inputs and the state is updated with what we expect to be sent over the wire 
3) ditto for DelimArray values

File: packages/run-it/src/components/LoginForm/LoginForm.tsx [hunk 1]
```diff
@@ -64,7 +64,9 @@ export const LoginForm: FC<LoginFormProps> = ({
 
   return (
     <Tooltip content={readyToLogin}>
-      <Button onClick={handleLogin}>Login</Button>
+      <Button key="loginButton" onClick={handleLogin}>
```
[the remark this answers is not in the document]
Comment (jkaster) [reply packages/run-it/src/components/LoginForm/LoginForm.tsx]: Removed key again

File: packages/run-it/src/components/RequestForm/formUtils.tsx [hunk 2]
```diff
@@ -61,13 +52,32 @@ const createDateItem = (
...
-  <div key={name}>
-    <Label>{name}</Label>
-    <InputDate
-      defaultValue={name in requestContent ? requestContent[name] : undefined}
-      onChange={handleChange.bind(null, name)}
-    />
-  </div>
+  <FormItem key={`${name}_fid`} id={name} label={name}>
+    <Popover
+      key={`${name}_pop`}
```
[the remark this answers is not in the document]
Comment (jkaster) [reply packages/run-it/src/components/RequestForm/formUtils.tsx:57]: Added myself. Didn't want to risk the Google cla bot issue

File: packages/api-explorer/src/scenes/MethodScene/utils.ts [hunk 3]
```diff
@@ -103,6 +106,13 @@ const createSampleBody = (spec: IApiModel, type: IType) => {
   return recurse(type)
 }
 
+const editType = (spec: IApiModel, type: IType) => {
```
[the remark this answers is not in the document]
Comment (jkaster) [reply packages/api-explorer/src/scenes/MethodScene/utils.ts:114]: I've added the comment but GithHub is currently refusing to display the update.

File: packages/run-it/src/components/RequestForm/formUtils.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (josephaxisa) [reply packages/run-it/src/components/RequestForm/formUtils.tsx:57]: Fair. It's definitely fixed now though, at least it had no complaints on my last PR

**Review by josephaxisa — COMMENTED** [review]
Looks good to me! Some design-related feedback that may be worth discussing is 
- the whitespace between form inputs and their labels and how it affects the form presentation when there are no inputs:
image
image
- toggle switches get "squashed" if there's a lengthy description:
toggle switches

File: packages/run-it/src/components/ShowResponse/responseUtils.tsx [hunk 4]
```diff
@@ -55,6 +54,12 @@ export const allSimple = (data: any[]) => {
   return true
 }
 
+const copyRaw = (code: string, language = 'unknown') => {
+  return (
+    <CodeCopy language={language} code={code} lineNumbers={false} transparent />
+  )
+}
+
```
[the remark this answers is not in the document]
Comment (jkaster) [reply packages/run-it/src/components/ShowResponse/responseUtils.tsx:62]: We could. I'd like to see how the transpiled code is different, if at all

**Comment by josephaxisa** [comment]
@jhardy do you have any thoughts about the above two points by any chance?

**Review by josephaxisa — APPROVED** [review]
caught up with John and we agreed to address any design feedback in a separate PR. This is ready to merge as is.

File: packages/run-it/src/components/ShowResponse/responseUtils.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jkaster) [reply packages/run-it/src/components/ShowResponse/responseUtils.tsx:62]: Looks like it's just stylistic so no reason to enforce a different syntax

**Pull request merged**